src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/volume/v2/AwsV2VolumePlugin.java

```diff
@@ -0,0 +1,147 @@
+				.availabilityZone(this.zone)
+				.build();
+		
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		CreateVolumeResponse volumeResponse = client.createVolume(volumeRequest);
+		String volumeId = volumeResponse.volumeId();
+		
+		CreateTagsRequest tagRequest = createVolumeTagName(volumeOrder, volumeId);
+		client.createTags(tagRequest);
+
+		return volumeId;
+	}
+
+	@Override
+	public VolumeInstance getInstance(VolumeOrder volumeOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE, volumeOrder.getInstanceId(), cloudUser.getToken()));
+
+		String volumeId = volumeOrder.getInstanceId();
+		DescribeVolumesRequest volumeRequest = DescribeVolumesRequest.builder()
+				.volumeIds(volumeId)
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		DescribeVolumesResponse volumeResponse = client.describeVolumes(volumeRequest);
+		VolumeInstance volumeInstance = mountVolumeInstance(volumeResponse);
+		return volumeInstance;
+	}
+
+	@Override
+	public void deleteInstance(VolumeOrder volumeOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE, volumeOrder.getInstanceId(), cloudUser.getToken()));
+		
+		String volumeId = volumeOrder.getInstanceId();
+		DeleteVolumeRequest volumeRequest = DeleteVolumeRequest.builder()
+				.volumeId(volumeId)
+				.build();
+		
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		try {
+			client.deleteVolume(volumeRequest);
+		} catch (Exception e) {
+			LOGGER.error(String.format(Messages.Error.ERROR_WHILE_REMOVING_RESOURCE, RESOURCE_NAME, volumeId), e);
+			throw new UnexpectedException();
+		}
+	}
+
+	protected VolumeInstance mountVolumeInstance(DescribeVolumesResponse response) throws InstanceNotFoundException {
+		if (!response.volumes().isEmpty()) {
+			Volume volume = response.volumes().get(FIRST_POSITION);
+			String id = volume.volumeId();
+			String cloudState = volume.stateAsString();
+			String name = volume.tags().get(FIRST_POSITION).value();
+			Integer size = volume.size();
+			return new VolumeInstance(id, cloudState, name, size);
+		}
+		throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+	}
+
+	protected CreateTagsRequest createVolumeTagName(VolumeOrder volumeOrder, String volumeId) {
+		String name = defineVolumeName(volumeOrder.getName());
+
+		Tag tagName = Tag.builder()
+				.key(AWS_TAG_NAME)
+				.value(name)
+				.build();
+
+		CreateTagsRequest tagRequest = CreateTagsRequest.builder()
+				.resources(volumeId)
+				.tags(tagName)
+				.build();
+
+		return tagRequest;
+	}
+	
+	protected String defineVolumeName(String volumeName) {
+		return volumeName == null ? SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + getRandomUUID() : volumeName;
+	}
+	
+	// This method is used to aid in the tests
+	protected String getRandomUUID() {
```

Comment:
Maybe we should have an util module to wrap this kind of functions, like ClientUtil. Let's think about it.

Reply:
I agree, this method also was used in other plugins from other clouds. Maybe this util module can be made available in the common so that it can be used by other clouds.